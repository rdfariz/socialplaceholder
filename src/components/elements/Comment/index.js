import React, { useState } from 'react';
import PropTypes from 'prop-types';
import EditCommentForm from '../../form/editComment';

function Component (props) {
  const { id, email, name, body, handleUpdateComment, handleDeleteComment } = props;

  const [isEditLayout, setIsEditLayout] = useState(false);

  const onSubmit = async (values) => {
    await handleUpdateComment(values, id);
  };

  const renderComment = (
    <div>
      <h2>by: <span className="font-semibold">{name}</span></h2>
      <p className="mt-1 text-sm">{body}</p>
      <div className="w-full flex flex-wrap mt-4">
        <button onClick={() => setIsEditLayout(true)} className="btn-edit text-sm border text-blue-600 rounded py-1 px-2 mr-2">Edit</button>
        <button onClick={() => handleDeleteComment(id)} className="btn-delete text-sm border text-red-600 rounded py-1 px-2 mr-2">Delete</button>
      </div>
    </div>
  );
  
  const renderFormEdit = (
    <div>
      <EditCommentForm initialValues={{ email, name, body }} onSubmit={onSubmit} />
    </div>
  );

  return (
    <div className="mb-3 pb-4 border-b">
      {isEditLayout ? renderFormEdit : renderComment}
    </div>
  );
};

Component.defaultProps = {
  name: '',
  id: '',
  postId: '',
  email: '',
  body: '',
  className: '',
  handleUpdateComment: () => {},
  handleDeleteComment: () => {}
};

Component.propTypes = {
  name: PropTypes.string,
  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  postId: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  email: PropTypes.string,
  body: PropTypes.string,
  className: PropTypes.string,
  handleUpdateComment: PropTypes.func,
  handleDeleteComment: PropTypes.func
};

export default Component;
