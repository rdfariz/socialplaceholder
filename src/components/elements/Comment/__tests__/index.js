import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Comment from '../';
import { shallow } from 'enzyme';

describe('src/components/elements/Comment', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(
      <Comment
        id="1"
        email="rfarizinsanp@gmail.com"
        name="tester"
        body="body"
        handleUpdateComment={jest.fn()}
        handleDeleteComment={jest.fn()}
      />
    );
    expect(tree).toMatchSnapshot();
  });

  test('should calling function delete when button delete clicked', () => {
    const onDelete = jest.fn();
    const wrapper = shallow(
      <Comment
        id="1"
        email="rfarizinsanp@gmail.com"
        name="tester"
        body="body"
        handleDeleteComment={onDelete}
      />
    );
    const btnDelete = wrapper.find('.btn-delete');
    btnDelete.first().simulate('click');
    expect(onDelete).toBeCalled();
  });
});
