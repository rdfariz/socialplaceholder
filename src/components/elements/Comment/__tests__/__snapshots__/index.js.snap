// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/Comment render 1`] = `
<div
  className="mb-3 pb-4 border-b"
>
  <div>
    <h2>
      by: 
      <span
        className="font-semibold"
      >
        tester
      </span>
    </h2>
    <p
      className="mt-1 text-sm"
    >
      body
    </p>
    <div
      className="w-full flex flex-wrap mt-4"
    >
      <button
        className="btn-edit text-sm border text-blue-600 rounded py-1 px-2 mr-2"
        onClick={[Function]}
      >
        Edit
      </button>
      <button
        className="btn-delete text-sm border text-red-600 rounded py-1 px-2 mr-2"
        onClick={[Function]}
      >
        Delete
      </button>
    </div>
  </div>
</div>
`;
